controllers/chat/initChatFromPost.js

```diff
@@ -42,35 +46,97 @@ const initChatFromPost = async (req, res) => {
+        postAnonUserInfo?.anon_user_info_emoji_code;
+      newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name =
+        postAnonUserInfo?.anon_user_info_emoji_name;
+
+      await ChatAnonUserInfo.create({
+        channel_id: newChannel.id,
+        my_anon_user_id: userModel.user_id,
+        target_user_id: targetUserModel.user_id,
+        anon_user_info_color_code: postAnonUserInfo?.anon_user_info_color_code,
+        anon_user_info_color_name: postAnonUserInfo?.anon_user_info_color_name,
+        anon_user_info_emoji_code: postAnonUserInfo?.anon_user_info_emoji_code,
+        anon_user_info_emoji_name: postAnonUserInfo?.anon_user_info_emoji_name
+      });
+    }
+
+    if (userModel.is_anonymous) {
+      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_code = color.code;
+      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_name = color.color;
+      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_code = emoji.emoji;
+      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_name = emoji.name;
+
+      await ChatAnonUserInfo.create({
+        channel_id: newChannel.id,
+        my_anon_user_id: userModel.user_id,
+        target_user_id: targetUserModel.user_id,
+        anon_user_info_color_code: color.code,
+        anon_user_info_color_name: color.color,
+        anon_user_info_emoji_code: emoji.emoji,
+        anon_user_info_emoji_name: emoji.name
+      });
+    }
+    //end add anon info to channel
 
     try {
-      if (!channel?.data?.name) {
+      if (!newChannel?.data?.name) {
         let channelType = CHANNEL_TYPE.CHAT;
         if (userModel.is_anonymous || targetUserModel.is_anonymous) {
           channelType = CHANNEL_TYPE.ANONYMOUS;
         }
 
-        await channel.updatePartial({
+        await newChannel.updatePartial({
           set: {
             channel_type: channelType,
-            name: [userModel?.username, targetUserModel?.username].join(', ')
+            name: [userModel?.username, targetUserModel?.username].join(', '),
+            anon_user_info_color_code: color.code,
+            anon_user_info_color_name: color.color,
+            anon_user_info_emoji_code: emoji.emoji,
+            anon_user_info_emoji_name: emoji.name,
+            better_channel_member: newStateMemberWithAnonInfo
           }
         });
       }
     } catch (e) {
       console.log(e);
     }
 
+    const channelState = await newChannel.watch();
+
     const response = {
-      ...createdChannel,
-      better_channel_members_object: channel.state.members,
-      better_channel_members: Object.values(channel.state.members)
+      ...channelState,
+      better_channel_members: Object.values(newStateMemberWithAnonInfo),
+      better_channel_members_objet: newStateMemberWithAnonInfo,
+      messageHistories: []
     };
 
+    await newChannel.stopWatching();
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There are a few issues in this section:

1. Error handling: There is no error handling for the `Promise.all` call at lines 31-35 and the `ChatAnonUserInfo.create` calls at lines 77-85 and 94-102. If any of these promises reject, it will cause an unhandled promise rejection which can lead to unpredictable behavior or crashes. It's recommended to wrap these calls in try-catch blocks.

2. Code duplication: The code for creating `ChatAnonUserInfo` is duplicated for `targetUserModel` and `userModel`. This could be abstracted into a separate function to adhere to the DRY (Don't Repeat Yourself) principle.

3. Security: The `newChannel.updatePartial` call at lines 113-123 updates the channel with user information. However, there doesn't seem to be any validation or sanitization of the user input before it's used here. This could potentially lead to security vulnerabilities such as Cross-Site Scripting (XSS) attacks if the user input contains malicious scripts.

Here are the suggested changes:

```diff
    const newChannel = client.channel('messaging', uuid(), {
       members
     });
    await newChannel.create();
    let newStateMemberWithAnonInfo = newChannel.state.members;

    //Add anon info to channel
    const addAnonInfoToChannel = async (userModel, color, emoji) => {
      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_code = color.code;
      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_color_name = color.color;
      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_code = emoji.emoji;
      newStateMemberWithAnonInfo[userModel.user_id].anon_user_info_emoji_name = emoji.name;

      try {
        await ChatAnonUserInfo.create({
          channel_id: newChannel.id,
          my_anon_user_id: userModel.user_id,
          target_user_id: targetUserModel.user_id,
          anon_user_info_color_code: color.code,
          anon_user_info_color_name: color.color,
          anon_user_info_emoji_code: emoji.emoji,
          anon_user_info_emoji_name: emoji.name
        });
      } catch (error) {
        console.error('Error creating ChatAnonUserInfo:', error);
      }
    };

    if (targetUserModel.is_anonymous) {
      const postAnonUserInfo = await PostAnonUserInfoFunction.checkSelfUsernameInPost(
        PostAnonUserInfo,
        User,
        {
          postId: postId,
          userId: targetUserModel.user_id
        },
        null,
        true
      );

      await addAnonInfoToChannel(targetUserModel, postAnonUserInfo, postAnonUserInfo);
    }

    if (userModel.is_anonymous) {
      await addAnonInfoToChannel(userModel, color, emoji);
    }
    //end add anon info to channel
 
     try {
      if (!newChannel?.data?.name) {
         let channelType = CHANNEL_TYPE.CHAT;
         if (userModel.is_anonymous || targetUserModel.is_anonymous) {
           channelType = CHANNEL_TYPE.ANONYMOUS;
         }
 
        // Sanitize user input before using it
        const sanitizedUsername1 = sanitize(userModel?.username);
        const sanitizedUsername2 = sanitize(targetUserModel?.username);

        await newChannel.updatePartial({
           set: {
             channel_type: channelType,
            name: [sanitizedUsername1, sanitizedUsername2].join(', '),
            anon_user_info_color_code: color.code,
            anon_user_info_color_name: color.color,
            anon_user_info_emoji_code: emoji.emoji,
            anon_user_info_emoji_name: emoji.name,
            better_channel_member: newStateMemberWithAnonInfo
           }
         });
       }
     } catch (e) {
       console.log(e);
     }
 
    const channelState = await newChannel.watch();

     const response = {
      ...channelState,
      better_channel_members: Object.values(newStateMemberWithAnonInfo),
      better_channel_members_objet: newStateMemberWithAnonInfo,
      messageHistories: []
     };
 
    await newChannel.stopWatching();

    await client.disconnectUser();

    return res.status(200).json(responseSuccess('sent', response));
```
Note: The `sanitize` function used in the diff is a placeholder for whatever sanitization method you choose to use. It could be a simple escape function or a more complex library depending on your needs.


<!-- This is an auto-generated comment by OSS CodeRabbit -->